openedx/core/djangoapps/credit/signals.py

```diff
@@ -33,25 +32,6 @@ def on_course_publish(course_key):
         log.info(u'Added task to update credit requirements for course "%s" to the task queue', course_key)
 
 
-@receiver(SignalHandler.pre_publish)
-def on_pre_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
-    """
-    Create user partitions for verification checkpoints.
-
-    This is a pre-publish step since we need to write to the course descriptor.
-    """
-    from openedx.core.djangoapps.credit import api
-    if api.is_credit_course(course_key):
```

Comment:
@cahrens this block being deleted is what's caused the SQL count to go up. The extra query, `QUERY = u'SELECT "credit_creditcourse"."id", "credit_creditcourse"."course_key", "credit_creditcourse"."enabled" FROM "credit_creditcourse" WHERE "credit_creditcourse"."enabled" = %s' - PARAMS = (True,)`, is executed in the line `is_credit = api.is_credit_course(course_id)` below in `listen_for_grade_calculation`, which is called when a `COURSE_GRADE_CHANGED` signal fires during our tests.

Previously, this pre_publish call was *also* executing `api.is_credit_course(course_key)`, which would make the same query as above earlier in the test (at course setup time) and cache it somehow to avoid another db hit. I was able to reproduce the query count increase by checking out master and changing this line to `if False:`

So the end result is that a query that was previously being made-and-cached during the un-counted portion of the test is now only being made during the counted portion of the test. @nasthagiri, is this a problem with how our tests are set up? Should we be trying to isolate our query counting from the signals that get called after the `recalculate_subsection_grade` task finishes?